android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java

```diff
@@ -737,7 +737,32 @@ public void removeAllChildren()
 			}
 		}
 	}
+	
+	/**
+	* Returns the view by the given ID.
+	* @module.api
+	*/
+	@Kroll.method
+	public TiViewProxy getViewById(String arg)
+	{
+		if (children != null) {
+			for (TiViewProxy child : children) {
+				if (child.children != null && child.children.size() > 0) {
+					TiViewProxy parentChild = child.getViewById(arg);
+					if (parentChild != null) {
+						return parentChild;
+					}
+				}
+
+				if (child.getProperty(TiC.PROPERTY_ID) && child.getProperty(TiC.PROPERTY_ID).equals(arg)) {
```

Comment:
Gotcha, changed it. But the original implementation was like the second suggestion above already (except the string-casting), that's why I wondered. Sorry for the confusion, let's finish this 🙂